script/run-comp-match

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+"""
+A script to run a competition match.
+"""
+
+import os
+import sys
+import shutil
+import argparse
+import tempfile
+import contextlib
+import subprocess
+from typing import Iterator
+from pathlib import Path
+from zipfile import ZipFile
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+import controller_utils  # isort:skip
+
+
+def get_zone_path(zone_id: int) -> Path:
+    robot_file: Path = controller_utils.get_zone_robot_file_path(zone_id)
+    return robot_file.parent
+
+
+def construct_match_data(args: argparse.Namespace) -> controller_utils.MatchData:
+    tlas = [
+        tla if tla != '-' else None
+        for tla in args.tla
+    ]
+
+    return controller_utils.MatchData(
+        args.match_num,
+        tlas,
+        args.duration,
+        recording_config=None,  # use the default
+    )
+
+
+@contextlib.contextmanager
+def temporary_arena_root(suffix: str) -> Iterator[None]:
+    arena_root = controller_utils.ARENA_ROOT
+    with tempfile.TemporaryDirectory(suffix=suffix) as tmpdir_name:
+        print(f"Using {tmpdir_name!r} as the arena")  # noqa:T001
+        os.environ['ARENA_ROOT'] = tmpdir_name
+        controller_utils.ARENA_ROOT = Path(tmpdir_name)
+        try:
+            yield
+        finally:
+            os.environ['ARENA_ROOT'] = str(arena_root)
+            controller_utils.ARENA_ROOT = arena_root
+
+
+def prepare_match(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    controller_utils.get_mode_file().write_text('comp\n')
+    controller_utils.record_match_data(match_data)
+
+    for zone_id, tla in enumerate(match_data.teams):
+        zone_path = get_zone_path(zone_id)
+
+        if zone_path.exists():
+            shutil.rmtree(zone_path)
+
+        if tla is None:
+            # no team in this zone
+            continue
+
+        zone_path.mkdir()
+        with ZipFile(f'{archives_dir / tla}.zip') as zipfile:
+            zipfile.extractall(zone_path)
+
+
+def run_match() -> None:
+    try:
+        subprocess.check_call([
+            'webots',
+            '--batch',
+            '--stdout',
+            '--stderr',
+            '--minimize',
+            str(REPO_ROOT / 'worlds' / 'Arena.wbt'),
```

Comment:
Do we want to explicitly set the simulation mode, after the first time-step the competition simulator takes over but this gets it started.
```suggestion
            '--mode=realtime',
            str(REPO_ROOT / 'worlds' / 'Arena.wbt'),
```

Reply:
Is this going to cause issues if the name changes, as we saw with 2021a?

Reply:
While technically yes, the change in 2021a was because of enable rendering was moved to a separate variable so it's unlikely the `realtime` would be renamed.